crates/env/src/engine/e2e/xts.rs

```diff
@@ -0,0 +1,343 @@
+// Copyright 2018-2022 Parity Technologies (UK) Ltd.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use super::{
+    log_info,
+    sr25519,
+    ContractExecResult,
+    ContractInstantiateResult,
+    IdentifyAccount,
+    Signer,
+    Verify,
+};
+use crate::Environment;
+
+use core::marker::PhantomData;
+use jsonrpsee::{
+    core::client::ClientT,
+    rpc_params,
+    ws_client::{
+        WsClient,
+        WsClientBuilder,
+    },
+};
+use sp_core::{
+    Bytes,
+    H256,
+};
+use subxt::{
+    tx::{
+        ExtrinsicParams,
+        TxEvents,
+    },
+    OnlineClient,
+};
+
+/// The gas limit for contract instantiate and call dry runs.
+const DRY_RUN_GAS_LIMIT: u64 = 500_000_000_000;
+
+// TODO(#xxx) Should be fetched automatically.
+#[subxt::subxt(runtime_metadata_path = "metadata/contracts-node.scale")]
+pub(super) mod api {}
```

Comment:
I don't see that being used anywhere.

Reply:
This is the recommended way of initializing `subxt` in a way that it generates the runtime types from the `*.scale` file: https://docs.rs/subxt/latest/subxt/#generating-runtime-types.

Reply:
Yes but you are not using it to generate the transaction. Instead you are doing this manually.